openvdb_houdini/CMakeLists.txt

```diff
@@ -24,111 +24,193 @@
 # IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
 # LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
 #
-cmake_minimum_required(VERSION 3.1)
-set(CMAKE_CXX_STANDARD 11)
+#[=======================================================================[
 
-PROJECT ( OpenVDB_Houdini )
+  CMake Configuration for the OpenVDB Houdini Plugin.
 
-OPTION ( OPENVDB_HOUDINI_SUBDIR "Install plugins to <prefix>/houdini/<version> instead of <prefix>/houdini<version>" OFF)
+  See the OpenVDBHoudiniSetup.cmake module for more information on how to
+  locate and use a Houdini Installation.
+
+  There are two main stages for the OpenVDB Houdini build; the OpenVDB
+  Houdini base shared library and all subsequent dependent nodes (SOPs,
+  SHOPs, procedurals etc.) By default, both are installed to the same
+  prefix:
+
+    <prefix>/include  - Shared openvdb_houdini headers
+    <prefix>/lib      - Location of libopenvdb_houdini
+    <prefix>/dso      - Location of all other .so
+
+  Where prefix is ${CMAKE_INSTALL_PREFIX}/houdini<version>. If you're
+  building the Houdini plug-ins independently of the Core OpenVDB library,
+  users may wish to set CMAKE_INSTALL_PREFIX to $HIH (provided by the
+  Houdini Environment).
+
+  If you wish to build and install the Houdini plug-in alongside the core
+  library, or simply wish to customize the install paths of the above,
+  the following variables can be set and provided to CMake:
+
+   OPENVDB_HOUDINI_INSTALL_LIB_BASE_DIR
+     - The prefix for the shared library and headers
+   OPENVDB_HOUDINI_INSTALL_BASE_DIR
+     - The prefix for all other dsos, icons, config files etcs.
+
+#]=======================================================================]
+
+PROJECT ( OpenVDBHoudini )
+CMAKE_MINIMUM_REQUIRED ( VERSION 3.3 )
+IF ( POLICY CMP0074 )
+  CMAKE_POLICY ( SET CMP0074 NEW )
+ENDIF ()
+
+# CMake options for installation paths (if not overriding OPENVDB_HOUDINI_INSTALL variables)
+OPTION ( OPENVDB_HOUDINI_SUBDIR "Install plugins to <prefix>/houdini/<version> instead of <prefix>/houdini<version>" OFF )
 OPTION ( OPENVDB_HOUDINI_SHORT_VERSION "User major.minor as <version> instead of major.minor.patch" ON )
-OPTION ( OPENVDB_HOUDINI_INSTALL_LIBRARY "Install headers and place the shared library in a lib directory." OFF)
-FIND_PACKAGE ( HDK REQUIRED )
+OPTION ( OPENVDB_HOUDINI_INSTALL_LIBRARY "Install headers and place the shared library in a lib directory." ON )
```

Comment:
Do we want any of these above options? We expose `OPENVDB_HOUDINI_INSTALL_LIB_BASE_DIR` and `OPENVDB_HOUDINI_INSTALL_BASE_DIR` to allow custom configuration of the houdini installation paths, and really these values should be set to the default of `houdini_get_default_install_dir` which is defined in the houdini cmake:

```
function ( houdini_get_default_install_dir output_var )

    set( _instdir "" )
    if ( _houdini_platform_linux )
        set( _instdir $ENV{HOME}/houdini${_houdini_release_version} )
    elseif ( _houdini_platform_osx )
        set( _instdir $ENV{HOME}/Library/Preferences/houdini/${_houdini_release_version} )
    elseif ( _houdini_platform_win )
        set( _instdir $ENV{HOMEDRIVE}$ENV{HOMEPATH}\\Documents\\houdini${_houdini_release_version} )
    else ()
        message( FATAL_ERROR "Invalid platform" )
    endif ()

    set( ${output_var} ${_instdir} PARENT_SCOPE )
endfunction ()
```

My opinion is we remove all three.

Reply:
@jmlait your thoughts here would be good. These granular options allow you to specify:
 - <prefix>/houdini17.0 vs <prefix>/houdini/17.0
 - houdini17.0 vs houdini17.0.456

I was thinking to remove these options, use the above cmake function in the Houdini install to get the default install dir and instead let users manually specify the installation directory if they need it to be changed.

Reply:
From my perspective the base install should be HOUDINI_USER_PREF_DIR, which is what that houdini_get_default_install_dir is generating.  It would seem that people who have their own pipelines will be able to just manually specify rather than trying to provide all the possible patterns automatically.  So I guess the question is where these came from?  I'd guess it is the needs of some studio, so they are better able to say whether they think they should be shipping defaults or not.

Reply:
These were there when the CMake was originally written by Nicholas Yue so I'm unsure on the use case at the moment (I'll see if I can send him an e-mail). I could get these values to work with `houdini_get_default_install_dir` but it would involve adding logic to manipulate the return value which I was hesitant to do as the functionality/result could change (however unlikely). In which case simply providing the ability to change the installation directory to whatever seems like the better solution.

Reply:
Spoke to Nicholas about this who agreed it would be good to use the Houdini CMake